File: src/main/java/com/gusrylmubarok/reddit/backend/service/PostService.java
package com.gusrylmubarok.reddit.backend.service;

import com.gusrylmubarok.reddit.backend.dto.request.PostRequest;
import com.gusrylmubarok.reddit.backend.dto.response.PostResponse;

import java.util.List;
import java.util.Optional;

public interface PostService {
    PostResponse save(PostRequest postRequest) throws Exception;
    PostResponse getPostById(Long id);
    List<PostResponse> getAllPosts();
    Optional<PostResponse> getPostsBySubreddit(Long id);
    Optional<PostResponse> getPostsByUsername(String username);
}

File: src/main/java/com/reddot/service/PostService.kt
package com.reddot.service

import com.reddot.model.PostRequest
import com.reddot.model.PostResponse

interface PostService {
    fun savePost(request: PostRequest) : PostResponse
    fun findAll() : List<PostResponse>
    fun findById(id: Long) : PostResponse
    fun findBySubreddit(id: Long) : List<PostResponse>
    fun findByUsername(username: String) : List<PostResponse>
}